src/liballoc/tests/string.rs

```diff
@@ -356,6 +356,25 @@ fn remove_bad() {
     "ศ".to_string().remove(1);
 }
 
+#[test]
+fn test_remove_matches() {
+    let mut s = "abc".to_string();
+
+    s.remove_matches('b');
+    assert_eq!(s, "ac");
+    s.remove_matches('b');
+    assert_eq!(s, "ac");
+
+    let mut s = "abcb".to_string();
+
+    s.remove_matches('b');
+    assert_eq!(s, "ac");
+
+    let mut s = "ศไทย中华Việt Nam; foobarศ".to_string();
+    s.remove_matches('ศ');
+    assert_eq!(s, "ไทย中华Việt Nam; foobar");
```

Comment:
here's a couple more tests that might be helpful. empty string input, and empty string result

```suggestion
    assert_eq!(s, "ไทย中华Việt Nam; foobar");

    let mut s = "".to_string();
    s.remove_matches('');
    assert_eq!(s, "");

    let mut s = "aaaaa".to_string();
    s.remove_matches('a');
    assert_eq!(s, "");
```

Reply:
I believe this is solved.